internal/resources/policy/kind/mutation/recipe/pod_security_schema.go

```diff
@@ -0,0 +1,533 @@
+		}
+	}
+
+	if v, ok := podSecurityData[SupplementalGroupsKey]; ok {
+		if v1, ok := v.([]interface{}); ok {
+			podSecuritySpec.SupplementalGroups = expandSupplementalGroups(v1)
+		}
+	}
+
+	return podSecuritySpec
+}
+
+func FlattenPodSecurity(podSecurity *policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1PodSecurity) (data []interface{}) {
+	if podSecurity == nil {
+		return data
+	}
+
+	flattenPodSecurity := make(map[string]interface{})
+
+	if podSecurity.AllowPrivilegeEscalation != nil {
+		flattenPodSecurity[AllowPrivilegeEscalationKey] = flattenAllowPrivilegeEscalation(podSecurity.AllowPrivilegeEscalation)
+	}
+
+	if podSecurity.CapabilitiesAdd != nil {
+		flattenPodSecurity[CapabilitiesAddKey] = flattenCapabilitiesAdd(podSecurity.CapabilitiesAdd)
+	}
+
+	if podSecurity.CapabilitiesDrop != nil {
+		flattenPodSecurity[CapabilitiesDropKey] = flattenCapabilitiesDrop(podSecurity.CapabilitiesDrop)
+	}
+
+	if podSecurity.FsGroup != nil {
+		flattenPodSecurity[FsGroupKey] = flattenFsGroup(podSecurity.FsGroup)
+	}
+
+	if podSecurity.Privileged != nil {
+		flattenPodSecurity[PrivilegedKey] = flattenPrivileged(podSecurity.Privileged)
+	}
+
+	if podSecurity.ReadOnlyRootFilesystem != nil {
+		flattenPodSecurity[ReadOnlyRootFilesystemKey] = flattenReadOnlyRootFilesystem(podSecurity.ReadOnlyRootFilesystem)
+	}
+
+	if podSecurity.RunAsGroup != nil {
+		flattenPodSecurity[RunAsGroupKey] = flattenRunAsGroup(podSecurity.RunAsGroup)
+	}
+
+	if podSecurity.RunAsNonRoot != nil {
+		flattenPodSecurity[RunAsNonRootKey] = flattenRunAsNonRoot(podSecurity.RunAsNonRoot)
+	}
+
+	if podSecurity.RunAsUser != nil {
+		flattenPodSecurity[RunAsUserKey] = flattenRunAsUser(podSecurity.RunAsUser)
+	}
+
+	if podSecurity.SeLinuxOptions != nil {
+		flattenPodSecurity[SeLinuxOptionsKey] = flattenSeLinuxOptions(podSecurity.SeLinuxOptions)
+	}
+
+	if podSecurity.SupplementalGroups != nil {
+		flattenPodSecurity[SupplementalGroupsKey] = flattenSupplementalGroups(podSecurity.SupplementalGroups)
+	}
+
+	return []interface{}{flattenPodSecurity}
+}
+
+func expandAllowPrivilegeEscalation(data []interface{}) *policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1PodSecurityRunAsUser {
+	return nil
+}
+
+func flattenAllowPrivilegeEscalation(allowPrivilegeEscalation *policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1PodSecurityAllowPrivilegeEscalation) (data []interface{}) {
+	if allowPrivilegeEscalation == nil {
+		return data
+	}
+
+	flattenAllowPrivilegeEscalation := make(map[string]interface{})
+
+	flattenAllowPrivilegeEscalation[ConditionKey] = *allowPrivilegeEscalation.Condition
+	flattenAllowPrivilegeEscalation[ValueKey] = *allowPrivilegeEscalation.Value
+
+	return []interface{}{flattenAllowPrivilegeEscalation}
+}
+
+func expandCapabilitiesAddKey(data []interface{}) *policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1PodSecurityCapabilitiesAdd {
```

Comment:
the function which just return `nil` is required?

Reply:
No. I just added proper implementation too.